src/lib/storage/storage_manager.cpp

```diff
@@ -260,4 +267,295 @@ std::ostream& operator<<(std::ostream& stream, const StorageManager& storage_man
+  ofstream.close();
+}
+
+// needed for attribute vector which is stored in a compact manner
+void export_compact_vector(const pmr_compact_vector& values, std::string file_name) {
+  //adapted to uint32_t format of later created map (see comment in `write_dict_segment_to_disk`)
+  export_value(static_cast<uint32_t>(values.bits()), file_name);
+  std::ofstream ofstream(file_name, std::ios::binary | std::ios::app);
+  ofstream.write(reinterpret_cast<const char*>(values.get()), static_cast<int64_t>(values.bytes()));
+  ofstream.close();
+}
+
+void export_compressed_vector(const CompressedVectorType type, const BaseCompressedVector& compressed_vector, 
+                              std::string file_name) {
+  switch (type) {
+    case CompressedVectorType::FixedWidthInteger4Byte:
+      export_values(dynamic_cast<const FixedWidthIntegerVector<uint32_t>&>(compressed_vector).data(), file_name);
+      return;
+    case CompressedVectorType::FixedWidthInteger2Byte:
+      export_values(dynamic_cast<const FixedWidthIntegerVector<uint16_t>&>(compressed_vector).data(), file_name);
+      return;
+    case CompressedVectorType::FixedWidthInteger1Byte:
+      export_values(dynamic_cast<const FixedWidthIntegerVector<uint8_t>&>(compressed_vector).data(), file_name);
+      return;
+    case CompressedVectorType::BitPacking:
+      export_compact_vector(dynamic_cast<const BitPackingVector&>(compressed_vector).data(), file_name);
+      return;
+    default:
+      Fail("Any other type should have been caught before.");
+  }
+}
+
+void StorageManager::write_dict_segment_to_disk(const std::shared_ptr<DictionarySegment<int>> segment, std::string file_name) {
+  /*
+   * Write dict segment to given file using the following format:
+   * 1. Number of elements in dictionary
+   * 2. Number of elements in attribute_vector
+   * 3. AttributeVectorCompressionID aka. size of int used in attribute vector
+   * 3. Dictionary values
+   * 4. Attribute_vector values
+   *
+   * For this exercise we assume an <int>-DictionarySegment with an FixedWidthIntegerVector<uint16_t> attribute_vector.
+   * As a next step we should use the AttributeVectorCompressionID to define the type of the FixedWidthIntegerVector
+   * and perhaps also write out the type of the DictionarySegment.
+   */
+
+  //TODO: Should this be continued?
+  // We will later mmap to an uint32_t vector/array. Therefore, we store all metadata points as uint32_t.
+  // This wastes up to three bytes of compression per metadata point but makes mapping much easier.
+  export_value(static_cast<uint32_t>(segment->dictionary()->size()), file_name);
+  export_value(static_cast<uint32_t>(segment->attribute_vector()->size()), file_name);
+
+  const auto compressed_vector_type_id = static_cast<uint32_t>(BinaryWriter::_compressed_vector_type_id<int>(*segment));
+  export_value(compressed_vector_type_id, file_name);
+
+  export_values<int32_t>(*segment->dictionary(), file_name);
```

Comment:
What if our segment does not use int32_t.
This is maybe a problem for later us.

Reply:
Lets give it to the later us :D 🥇

Reply:
true